examples/cfq/syntax-based/grammar.py

```diff
@@ -74,29 +77,98 @@ def generate_grammar(grammar_str: str):
     rules.append(current_rule)
   sub_rules = []
   for rule in rules:
-    sub_rules += generate_sub_rules(rule)
+    sub_rules += extract_sub_rules_from_rules(rule)
+  return sub_rules
+
+class TermType(Enum):
+  SYNTAX_TERM = 0
```

Comment:
Nit: Maybe call this `STRING_TERM`.

Reply:
why? all other terms are strings. Wouldn't it make it more confusing?

Reply:
Hmm yes that is true. I was thinking you could call this something like `LITERAL_STRING`, which means it is just a string in double quotes, but no need to change it, I was just a detail.